src/main/kotlin/org/rust/lang/core/resolve/RustResolveEngine.kt

```diff
@@ -150,6 +155,19 @@ object RustResolveEngine {
         return resolveIn(ResolveScopeVisitor(ref, visited), enumerateScopesFor(ref))
     }
 
+    private fun resolveImportSpec(ref: RustImportSpec, visited: MutableSet<RustUseItem>): ResolveResult {
+        val basePath = ref.parentOfType<RustUseItem>()?.viewPath?.pathPart ?: return ResolveResult.UNRESOLVED
+
+        // NOTE(matklad): this is not necessary a module
```

Comment:
`NOTE` (word) itself makes it a bit clunky, but comments are OK (and more than that: absolutely vital in some non-trivial cases such as this one).

Code snippets would be great.